zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/RoleTest.java

```diff
@@ -52,4 +52,61 @@ public void shouldNotDuplicate() {
                 + name
                 + "', but a role with this name already exists");
   }
+
+  @Test
+  public void shouldUpdateRole() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var createdRecord = ENGINE.role().newRole(name).create();
+
+    // when
+    final var newName = UUID.randomUUID().toString();
+    final var updatedRoleRecord =
+        ENGINE.role().updateRole(createdRecord.getValue().getRoleKey()).withName(newName).update();
+
+    final var updatedRole = updatedRoleRecord.getValue();
+    Assertions.assertThat(updatedRole).isNotNull().hasFieldOrPropertyWithValue("name", newName);
+  }
+
+  @Test
+  public void shouldRejectIfRoleIsNotPresent() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var roleRecord = ENGINE.role().newRole(name).create();
+
+    // when
+    final var notPresentRoleKey = 1L;
+    final var notPresentUpdateRecord =
+        ENGINE.role().updateRole(notPresentRoleKey).expectRejection().update();
+
+    final var createdRole = roleRecord.getValue();
+    Assertions.assertThat(createdRole).isNotNull().hasFieldOrPropertyWithValue("name", name);
+
+    assertThat(notPresentUpdateRecord)
+        .hasRejectionType(RejectionType.NOT_FOUND)
+        .hasRejectionReason(
+            "Expected to update role with key '"
+                + notPresentRoleKey
+                + "', but a role with this key does not exists");
+  }
+
+  @Test
+  public void shouldRejectIfRoleWithSameNameIsPresent() {
```

Comment:
🔧 Should we also test for a rejection if a user is not authorized? Or is it enough that the authorization behavior class is tested?